addons/mail/static/src/core/web/chatter.js

```diff
@@ -361,8 +369,9 @@ export class Chatter extends Component {
     async scheduleActivity() {
         this.closeSearch();
         const schedule = async (threadId) => {
+            const thread = this.store.Thread.get({ model: this.props.threadModel, id: threadId });
```

Comment:
It's not `this.props` here, the `onNextUpdate` part is actually using `nextProps`, `schedule` should receive `thread` directly rather than id.

Reply:
I would prefer to pass `threadModel` and `threadId` to `schedule` and parse  to the `thread` model inside so that there is less redutant code as `this.threadService.getThread(threadModel, threadId);` everywhere to get the model when calling `schedule`